### PR TITLE
Feature/rework user location

### PR DESCRIPTION
**Pull request by MariyaSin**
-add latitude, longitude, WorkHours to edit user request.
-remove Patch mappers to Patch directory
-update fined user response
-update Image table name to UsersAvatars

### CODE REVIEW

File: src/UserService.Business/Commands/User/EditUserCommand.cs [hunk 1]
```diff
@@ -23,61 +27,88 @@ namespace LT.DigitalOffice.UserService.Business.Commands.User
...
+        o => o.path.EndsWith(nameof(EditUserRequest.Latitude), StringComparison.OrdinalIgnoreCase)) is not null;
+      bool longitudeOperation = patch.Operations.FirstOrDefault(
+        o => o.path.EndsWith(nameof(EditUserRequest.Longitude), StringComparison.OrdinalIgnoreCase)) is not null;
+
+      if ((latitudeOperation || longitudeOperation) && !(latitudeOperation && longitudeOperation))
       {
-        return _responseCreater.CreateFailureResponse<bool>(HttpStatusCode.Forbidden);
+        return _responseCreator.CreateFailureResponse<bool>(
+          HttpStatusCode.BadRequest,
+          new() { "latitude and longitude must have a value." });
```
Comment (Cheephot) [inline src/UserService.Business/Commands/User/EditUserCommand.cs:104]: Did you say that don't need a description of the error

File: src/UserService.Mappers/Patch/PatchDbUserCertificateMapper.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+namespace LT.DigitalOffice.UserService.Patch.Models
+{
+  public class PatchDbUserCertificateMapper : IPatchDbUserCertificateMapper
+  {
+    public JsonPatchDocument<DbUserCertificate> Map(JsonPatchDocument<EditCertificateRequest> request, Guid? imageId)
+    {
+      if (request == null)
+      {
+        throw new ArgumentNullException(nameof(request));
```
Comment (Cheephot) [inline src/UserService.Mappers/Patch/PatchDbUserCertificateMapper.cs:17]: return null?

File: src/UserService.Mappers/Patch/PatchDbUserCommunicationMapper.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+namespace LT.DigitalOffice.UserService.Patch.Models
+{
+  public class PatchDbUserCommunicationMapper : IPatchDbUserCommunicationMapper
+  {
+    public JsonPatchDocument<DbUserCommunication> Map(JsonPatchDocument<EditCommunicationRequest> request)
+    {
+      if (request == null)
+      {
+        throw new ArgumentNullException(nameof(request));
```
Comment (Cheephot) [inline src/UserService.Mappers/Patch/PatchDbUserCommunicationMapper.cs:17]: same

File: src/UserService.Mappers/Patch/PatchDbUserEducationMapper.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace LT.DigitalOffice.UserService.Patch.Models
+{
+  public class PatchDbUserEducationMapper : IPatchDbUserEducationMapper
+  {
+    public JsonPatchDocument<DbUserEducation> Map(JsonPatchDocument<EditEducationRequest> request)
+    {
+      if (request == null)
+      {
+        throw new ArgumentNullException("Invalid request value");
```
Comment (Cheephot) [inline src/UserService.Mappers/Patch/PatchDbUserEducationMapper.cs:17]: same

File: src/UserService.Business/Commands/Image/CreateImageCommand.cs [hunk 5]
```diff
@@ -135,7 +135,7 @@ public class CreateImageCommand : ICreateImageCommand
 
       if (response.Body != null)
       {
-        DbEntityImage dbEntityImage = _dbEntityImageMapper.Map(response.Body.Value, request.EntityId, request.IsCurrentAvatar);
+        DbUserAvatar dbEntityImage = _dbEntityImageMapper.Map(response.Body.Value, request.EntityId, request.IsCurrentAvatar);
```
Comment (ndoriomedov) [inline src/UserService.Business/Commands/Image/CreateImageCommand.cs:138]: rename variable too maybe?

File: src/UserService.Business/Commands/User/EditUserCommand.cs [hunk 6]
```diff
@@ -23,61 +27,88 @@ namespace LT.DigitalOffice.UserService.Business.Commands.User
...
   {
+    private readonly IEditUserRequestValidator _validator;
     private readonly IUserRepository _userRepository;
     private readonly IUserCredentialsRepository _credentialsRepository;
+    private readonly IUserLocationRepository _locationRepository;
     private readonly IPatchDbUserMapper _mapperUser;
     private readonly IAccessValidator _accessValidator;
     private readonly IHttpContextAccessor _httpContextAccessor;
-    private readonly IResponseCreater _responseCreater;
+    private readonly IResponseCreater _responseCreator;
```
Comment (ndoriomedov) [inline src/UserService.Business/Commands/User/EditUserCommand.cs:37]: update kernel

File: src/UserService.Data/UserLocationRepository.cs [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+    private async Task<bool> CreateAsync(Guid userId, JsonPatchDocument<DbUserLocation> patch)
+    {
+      DbUserLocation dbUserLocation = new()
+      {
+        Id = Guid.NewGuid(),
+        UserId = userId,
+        CreatedAtUtc = DateTime.Now,
+        CreatedBy = _httpContextAccessor.HttpContext.GetUserId()
+      };
```
Comment (ndoriomedov) [inline src/UserService.Data/UserLocationRepository.cs:34]: maybe do it in mapper?

File: src/UserService.Data/UserRepository.cs [hunk 8]
```diff
@@ -154,7 +154,7 @@ select new
          .Select(x =>
          {
            DbUser user = x.Select(x => x.User).FirstOrDefault();
-           Guid? id = x.Select(x => x.EntityImage).FirstOrDefault(x => x != null && x.EntityId == user.Id && x.IsCurrentAvatar)?.ImageId;
+           Guid? id = x.Select(x => x.EntityImage).FirstOrDefault(x => x != null && x.UserId == user.Id && x.IsCurrentAvatar)?.ImageId;
```
Comment (ndoriomedov) [inline src/UserService.Data/UserRepository.cs:157]: rename entityImage

File: src/UserService.Mappers/Db/DbEntityImageMapper.cs [hunk 9]
```diff
@@ -9,7 +9,7 @@ namespace LT.DigitalOffice.UserService.Mappers.Db
 {
   public class DbEntityImageMapper : IDbEntityImageMapper
```
Comment (ndoriomedov) [inline src/UserService.Mappers/Db/DbEntityImageMapper.cs:10]: mb rename?

File: src/UserService.Mappers/Db/DbEntityImageMapper.cs [hunk 10]
```diff
@@ -18,20 +18,20 @@ public List<DbEntityImage> Map(List<Guid> imagesIds, CreateImageRequest request)
...
       {
         Id = Guid.NewGuid(),
-        EntityId = entityId,
+        UserId = entityId,
         ImageId = x
       }).ToList();
     }
 
-    public DbEntityImage Map(Guid imageId, Guid entityId, bool isCurrentAvatar = false)
+    public DbUserAvatar Map(Guid imageId, Guid entityId, bool isCurrentAvatar = false)
```
Comment (ndoriomedov) [inline src/UserService.Mappers/Db/DbEntityImageMapper.cs:29]: guid userId

File: src/UserService.Mappers/Db/DbEntityImageMapper.cs [hunk 11]
```diff
@@ -18,20 +18,20 @@ public List<DbEntityImage> Map(List<Guid> imagesIds, CreateImageRequest request)
 
       Guid entityId = request.EntityId;
```
Comment (ndoriomedov) [inline src/UserService.Mappers/Db/DbEntityImageMapper.cs:19]: userId, rename it in CreateImageRequest too

File: src/UserService.Mappers/Db/DbUserMapper.cs [hunk 12]
```diff
@@ -1,99 +1,97 @@
...
-        {
-            if (request == null)
-            {
-                throw new ArgumentNullException(nameof(request));
-            }
+    public DbUser Map(ICreateAdminRequest request)
+    {
+      if (request == null)
+      {
+        throw new ArgumentNullException(nameof(request));
```
Comment (ndoriomedov) [inline src/UserService.Mappers/Db/DbUserMapper.cs:66]: don't throw

File: src/UserService.Mappers/Db/Interfaces/IDbEntityImageMapper.cs [hunk 13]
```diff
@@ -9,7 +9,7 @@ namespace LT.DigitalOffice.UserService.Mappers.Db.Interfaces
   [AutoInject]
   public interface IDbEntityImageMapper
   {
-    List<DbEntityImage> Map(List<Guid> imagesIds, CreateImageRequest request);
-    DbEntityImage Map(Guid imageId, Guid entityId, bool isCurrentAvatar = false);
+    List<DbUserAvatar> Map(List<Guid> imagesIds, CreateImageRequest request);
+    DbUserAvatar Map(Guid imageId, Guid entityId, bool isCurrentAvatar = false);
```
Comment (ndoriomedov) [inline src/UserService.Mappers/Db/Interfaces/IDbEntityImageMapper.cs:13]: guid userId

File: src/UserService.Validation/User/CreateUserRequestValidator.cs [hunk 14]
```diff
@@ -54,42 +51,31 @@ public class CreateUserRequestValidator : AbstractValidator<CreateUserRequest>,
...
 
       RuleFor(user => user.Status)
         .IsInEnum().WithMessage("Wrong status value.");
 
+      When(
+        u => u.Rate is not null,
+        () =>
+          RuleFor(user => user.Rate)
+            .GreaterThan(0)
+            .LessThanOrEqualTo(1));
```
Comment (ndoriomedov) [inline src/UserService.Validation/User/CreateUserRequestValidator.cs:71]: add messages mb?

**Pull request closed without merging**